### ISSUE TITLE
Support usage of alternative Storage backends/providers

### ISSUE THREAD

**Issue by simonsan** [stingle/stingle-photos-android#47]
I'm using Nextcloud to store and sync my pictures. Besides not being **locked in** to the services with a user account (#46) it would be really cool, if we could get `alternative storage providers` to chose from like Nextcloud (maybe also Dropbox, Google Drive, OneDrive) to save our pictures to.

**Comment by alexamiryan** [comment]
I don't think this is something we would implement, mainly because it's impossible to offer seamless sharing and sync functionality as we have now with third party storage providers. We need to have control over the files and our databases to offer current functionality. Right now you can share an album with tens of thousands photos in it with your friend and it would be just a matter of doing several cryptographic operations and transferring several kilobytes of data. This kind of experience would be impossible with third party storage providers and frankly it defeats the purpose of Stingle Photos. 
Stingle Photos is meant to be one stop, all in one secure gallery and sync app and if user needs to take care of other providers that's not an experience that we are building.

**Comment by acme4j** [comment]
Do you guys have any plans to disclose source code of backend?

**Comment by alexamiryan** [comment]
We don't have such plans for now

**Comment by alexanderadam** [comment]
@simonsan maybe you rather want to use [rcx](https://f-droid.org/en/packages/io.github.x0b.rcx/) for that. It is free software based on [rclone (thus allowing to use encrypted access to Nextcloud, and various other storages)](https://rclone.org/webdav/#nextcloud).

It's not a nice dedicated photo solution like stingle though and [doesn't have a functionality to automatically upload photos (making it a very manual task right now)](https://github.com/x0b/rcx/issues/128).

**Comment by alexamiryan** [comment]
We are in the process of making the server open-source and self hostable

**Comment by poVoq** [comment]
There is actually an alternative back-end implementation already: https://c2fmzq.org/

They also have modified to allow changing the back-end URL: https://github.com/c2FmZQ/stingle-photos-for-self-hosted-server

But it would be nice if this could be supported by the official apps.

Thanks!

**Comment by alexamiryan** [comment]
That's interesting. The funny part is that we are few days away from making official server code open source and making it self-hostable. Currently working on making it really easy, like running a single docker command.

**Comment by alexamiryan** [comment]
> There is actually an alternative back-end implementation already: https://c2fmzq.org/
> 
> They also have modified the client to allow changing the back-end URL: https://github.com/c2FmZQ/stingle-photos-for-self-hosted-server
> 
> But it would be nice if this could be supported by the official apps.
> 
> Thanks!

That's interesting. The funny part is that we are few days away from making official server code open source and making it self-hostable. Currently working on making it really easy, like running a single docker command.

**Comment by poVoq** [comment]
Any news on the open-sourcing of the back end?
